src/operations/update.ts

```diff
@@ -142,26 +153,11 @@ export class UpdateOneOperation extends UpdateOperation {
     }
   }
 
-  override executeCallback(
+  override async execute(
     server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<UpdateResult | Document>
-  ): void {
-    super.executeCallback(server, session, (err, res) => {
-      if (err || !res) return callback(err);
-      if (this.explain != null) return callback(undefined, res);
-      if (res.code) return callback(new MongoServerError(res));
-      if (res.writeErrors) return callback(new MongoServerError(res.writeErrors[0]));
-
-      callback(undefined, {
-        acknowledged: this.writeConcern?.w !== 0 ?? true,
-        modifiedCount: res.nModified != null ? res.nModified : res.n,
-        upsertedId:
-          Array.isArray(res.upserted) && res.upserted.length > 0 ? res.upserted[0]._id : null,
-        upsertedCount: Array.isArray(res.upserted) && res.upserted.length ? res.upserted.length : 0,
-        matchedCount: Array.isArray(res.upserted) && res.upserted.length > 0 ? 0 : res.n
-      });
-    });
+    session: ClientSession | undefined
+  ): Promise<UpdateResult | Document> {
+    return super.execute(server, session);
```

Comment:
You can remove this and any other method that just passes through the arguments to the same method on its parent class.